examples/omni-call/contracts/interfaces/IOmniCall.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.22;
+
+/// -----------------------------------------------------------------------
+/// Imports
+/// -----------------------------------------------------------------------
+
+//  ==========  LayerZero imports    ==========
+
+import { MessagingFee } from "@layerzerolabs/oapp-evm/contracts/oapp/OApp.sol";
+import { MessagingReceipt } from "@layerzerolabs/oapp-evm/contracts/oapp/OAppSender.sol";
+
+//  ==========  Internal imports    ==========
+
+import { Call, Transfer } from "../OmniCallMsgCodecLib.sol";
+
+/// -----------------------------------------------------------------------
+/// Interface
+/// -----------------------------------------------------------------------
+
+/**
+ * @title IOmniCall
+ * @author LayerZeroLabs (@EWCunha).
+ * @notice Interface for the OmniCall contract.
+ */
+interface IOmniCall {
+    /// -----------------------------------------------------------------------
+    /// Custom errors
+    /// -----------------------------------------------------------------------
+
+    /// @dev Error for when a zero gas limit is passed as a parameter.
+    error LZ_OmniCall__ZeroGasLimit();
+
+    /// @dev Error for when a call target is the endpoint.
+    error LZ_OmniCall__InvalidTarget();
+
+    /// @dev Error for when a call has no calldata and non-zero value.
+    error LZ_OmniCall__InvalidCall();
+
+    /// -----------------------------------------------------------------------
+    /// State-change public/external functions
+    /// -----------------------------------------------------------------------
+
+    /**
+     * @notice Sends a message and/or transfers gas tokens.
+     * @notice UNSAFE: there is the possibility the transaction will revert on the destination chain.
+     * To avoid that, be sure to provide correct `dstGasLimit`.
+     * @param messageType: type of message to be sent;
+     * @param dstEid: endpoint ID of the destination chain;
+     * @param dstCall: destination call;
+     * @param dstTransfer: destination transfer;
+     * @param dstGasLimit: gas limit for destination call/transfer.
+     */
+    function send(
```

Comment:
Also a fun idea to explore, see if we can add composeMsg to this. Then we can achieve horizontal composability by building an OmniComposer 

And sending arbitrary composer calls